### DISCUSSION TITLE
Run shutdown script on exit (equivalent to autostart on startup)

### DISCUSSION THREAD

**Discussion by guzz46** [labwc/labwc#1804] [General]
Does anyone know what that option is? it says it was added in 0.7.2, does labwc have the option to run a script on exit now?

**Comment by Consolatis** [comment]
https://labwc.github.io/labwc-config.5.html

> The shutdown file is executed as a shell script when labwc is preparing to terminate itself. All environment variables, including WAYLAND_DISPLAY and DISPLAY, will be available to the script. However, because the script runs asynchronously with other termination tasks, the shutdown file should not assume that the display will be usable. This file is useful to perform any custom operations necessary to finalize a labwc session.

**Reply by guzz46** [reply]
Thanks, I didn't realise there was a shutdown file, last time I checked there wasn't one.

**Reply by Consolatis** [reply]
It's new in 0.7.2. Maybe we should add an empty `shutdown` file with a comment along with the `autostart` file, that should make it easier to discover than by reading through the man pages.

**Reply by guzz46** [reply]
That's probably a good idea, I was thinking it might come in handy for me because I run my browser in tmpfs and I have a script that copies it to a backup folder when I logout/reboot/shutdown, but it sounds like it runs the shutdown script when you exit labwc, which would work if I logged out, but I use a rofi powermenu that makes use of systemctl reboot, so I guess it probably wouldn't run if I rebooted from within labwc, unless maybe I changed the powermenu to perform 'labwc -e && systemctl reboot'

**Comment by stefonarch** [comment]
> It's new in 0.7.2. Maybe we should add an empty `shutdown` file with a comment along with the `autostart` file, that should make it easier to discover than by reading through the man pages.

I think that's a good idea, just maybe add a comment in it.

**Reply by Consolatis** [reply]
- #1809
